### DISCUSSION TITLE
Worker nodes become unavailable when killing master node (HA)

### DISCUSSION THREAD

**Discussion by boedy** [k3s-io/k3s#10453] [Q&A]
I've encountered an issue in my Kubernetes high-availability (HA) setup and would appreciate some insights.

**Issue**:
When I kill one of the master nodes in my HA setup, I'm noticing that some of my worker nodes become temporarily unavailable, and as a result, some pods are being redeployed.

**Details**:
- **Kubernetes Version**: v1.25.6
- **Cluster Setup**:
    - 3 master nodes with external MySQL datastore
    - Fixed Registration address through Load balancer in front of master nodes (Round Robin)
    - 8 worker nodes
    - Kubelet args:
    ```
     --kubelet-arg "node-status-update-frequency=4s" \
     --kubelet-arg "eviction-hard=imagefs.available<15%,memory.available<100Mi,nodefs.available<5Gi"
    ```

**Questions**:
1. Is it expected behavior for worker nodes to become temporarily unavailable when a master node in an HA setup goes down?
2. Should pods be redeployed in this scenario, or should they continue running without interruption?
3. What might be causing this behavior, and how can it be mitigated?

I was under the impression that an HA setup should prevent such disruptions and maintain continuous availability of the worker nodes and pods.

Any guidance or suggestions would be greatly appreciated! Thanks

**Comment by brandond** [comment]
v1.25 is end of life. Can you test on an actively maintained version of k3s, and confirm if you still see the same behavior?
